### PR TITLE
DIA-4611 integrate custom consent call into Android SDK

### PR DESCRIPTION
**Pull request by Nevazhnovu**
[DIA-4611](https://sourcepoint.atlassian.net/browse/DIA-4611) [Android] Integrate custom-consent call

[DIA-4611]: https://sourcepoint.atlassian.net/browse/DIA-4611?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: cmplibrary/src/main/java/com/sourcepoint/cmplibrary/SpConsentLibImpl.kt [hunk 1]
```diff
@@ -254,6 +253,14 @@ internal class SpConsentLibImpl(
                                 msg = "${ccResp.t.message}",
```
Comment (andresilveirah) [inline cmplibrary/src/main/java/com/sourcepoint/cmplibrary/SpConsentLibImpl.kt:253]: Let's get rid of this call to logger.

File: cmplibrary/src/main/java/com/sourcepoint/cmplibrary/SpConsentLibImpl.kt [hunk 2]
```diff
@@ -267,16 +274,15 @@ internal class SpConsentLibImpl(
...
-            propertyId = campaignManager.spConfig.propertyId,
-            categories = categories,
-            legIntCategories = legIntCategories,
-            vendors = vendors
-        )
         executor.run {
             executeOnWorkerThread {
-                val ccResp = service.deleteCustomConsentToServ(customConsentReq, env)
+                val ccResp = service.deleteCustomConsentToServ(
+                    consentUUID = campaignManager.gdprUuid ?: "",
```
Comment (andresilveirah) [inline cmplibrary/src/main/java/com/sourcepoint/cmplibrary/SpConsentLibImpl.kt:280]: Same here, regarding empty uuid.

File: cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt [hunk 3]
```diff
@@ -69,67 +65,76 @@ internal class ServiceImpl(
...
+                    e -> e.key to GDPRPurposeGrants(e.value.vendorGrant,e.value.purposeGrants)
+            }.toMap()
+            campaignManager.gdprConsentStatus = campaignManager.gdprConsentStatus?.copy(
+                grants = grants,
+                categories = response.categories,
+                vendors = response.vendors,
+                legIntCategories = response.legIntCategories,
+                specialFeatures = response.specialFeatures
+            )
+            campaignManager.gdprConsentStatus!!
```
Comment (andresilveirah) [inline cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt:98]: What's the purpose of this unwrapping in line 98?
```kotlin
campaignManager.gdprConsentStatus!!
```

If you want to return it, it's better to do it explicitly with the `return` keyword.

File: cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt [hunk 4]
```diff
@@ -69,67 +65,76 @@ internal class ServiceImpl(
...
```
Comment (andresilveirah) [inline cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt:134]: Same here regarding the unwrapping.

File: cmplibrary/src/main/java/com/sourcepoint/cmplibrary/SpConsentLibImpl.kt [hunk 5]
```diff
@@ -234,16 +234,15 @@ internal class SpConsentLibImpl(
...
-            propertyId = campaignManager.spConfig.propertyId,
-            categories = categories,
-            legIntCategories = legIntCategories,
-            vendors = vendors
-        )
         executor.run {
             executeOnWorkerThread {
-                val ccResp = service.sendCustomConsentServ(customConsentReq, env)
+                val ccResp = service.sendCustomConsentServ(
+                    consentUUID = campaignManager.gdprUuid ?: "",
```
[the remark this answers is not in the document]
Comment (Nevazhnovu) [reply cmplibrary/src/main/java/com/sourcepoint/cmplibrary/SpConsentLibImpl.kt:240]: The funny thing is that I took the approach that was used in the old version of code. 
My motivation was: it definitely worked this way therefore no need to change this code. 
Will change!

**Pull request merged**